velox/common/base/SimdUtil-inl.h

```diff
@@ -1068,6 +1068,73 @@ struct HalfBatchImpl<
   using Type = Batch64<T>;
 };
 
+template <typename T, typename U, typename A>
+struct ReinterpretBatch {
+#if XSIMD_WITH_SSE2
+  static xsimd::batch<T, A> apply(xsimd::batch<U, A> data, const xsimd::sse2&) {
+    return xsimd::batch<T, A>(data);
+  }
+#endif
+
+#if XSIMD_WITH_AVX
+  static xsimd::batch<T, A> apply(xsimd::batch<U, A> data, const xsimd::avx&) {
+    return xsimd::batch<T, A>(data);
+  }
+#endif
+};
+
+template <typename T, typename A>
+struct ReinterpretBatch<T, T, A> {
+  static xsimd::batch<T, A> apply(xsimd::batch<T, A> data, const A&) {
+    return data;
+  }
+};
+
+#if XSIMD_WITH_NEON
+
+template <typename A>
+struct ReinterpretBatch<uint8_t, int8_t, A> {
+  static xsimd::batch<uint8_t, A> apply(
+      xsimd::batch<int8_t, A> data,
+      const xsimd::neon&) {
+    return vreinterpret_u8_s8(data.data);
+  }
+};
+
+template <typename A>
+struct ReinterpretBatch<uint16_t, int16_t, A> {
+  static xsimd::batch<uint16_t, A> apply(
+      xsimd::batch<int16_t, A> data,
+      const xsimd::neon&) {
+    return vreinterpret_u16_s16(data.data);
```

Comment:
You need to change these to `vreinterpretq`.

Reply:
Oops, I need to define another set of implementations for NEON64, thanks for the catch